File: game/scripts/vscripts/modifiers/upgrades/items/modifier_upgrade_item_mithril_hammer.lua
modifier_upgrade_item_mithril_hammer = class({})

function modifier_upgrade_item_mithril_hammer:RemoveOnDeath() return false end
function modifier_upgrade_item_mithril_hammer:IsPurgable() return false end

function modifier_upgrade_item_mithril_hammer:OnCreated()
    self.damage_per_stack = 4
end

function modifier_upgrade_item_mithril_hammer:DeclareFunctions()
    return { 
        MODIFIER_PROPERTY_PREATTACK_BONUS_DAMAGE,
    }
end

function modifier_upgrade_item_mithril_hammer:GetModifierPreAttack_BonusDamage(params)
    return self.damage_per_stack
end

function modifier_upgrade_item_mithril_hammer:GetTexture(params)
    return 'item_mithril_hammer'
end

File: src/vscripts/modifiers/upgrades/items/modifier_upgrade_item_mithril_hammer.ts
import { CustomModifier } from "../../../abilities/framework/custom_modifier";
import { registerModifier } from "../../../lib/dota_ts_adapter";

@registerModifier("modifier_upgrade_item_mithril_hammer")
class ModifierItemMithrilHammer extends CustomModifier {
    damagePerStack = 4;

    DeclareFunctions() {
        return [ModifierFunction.PREATTACK_BONUS_DAMAGE];
    }

    GetModifierPreAttack_BonusDamage() {
        return this.damagePerStack;
    }

    GetTexture() {
        return "item_mithril_hammer";
    }
}
